scripts/portstat

```diff
@@ -140,13 +140,12 @@ class Portstat(object):
         """
         # Get speed from APPL_DB
         full_table_id = PORT_STATUS_TABLE_PREFIX + port_name
-        speed = PORT_RATE
         for ns in self.multi_asic.get_ns_list_based_on_options():
             self.db = multi_asic.connect_to_all_dbs_for_ns(ns)
             speed = self.db.get(self.db.APPL_DB, full_table_id, PORT_SPEED_FIELD)
             if speed is not None:
                 return int(speed)//1000
-        return speed
+        return PORT_RATE
```

Comment:
do not understand the fix here. why does this fix problem?

Reply:
it looks like a personal preference, you do not need to change the code here.

Reply:
This fix is nothing to do with issue reported.
I was getting below Jenkins error, hence the code change here to fix error.
Variable "port_rate" was updated from "get_port_speed()"

18:22:17  tests/portstat_test.py:227: AssertionError
18:22:17  ----------------------------- Captured stdout call -----------------------------
18:22:17  1
18:22:17  The rates are calculated within 3 seconds period
18:22:17  
18:22:17  ----------------------------- Captured stderr call -----------------------------
18:22:17  Traceback (most recent call last):
18:22:17    File "/sonic/sonic-utilities/scripts/portstat", line 464, in <module>
18:22:17      main()
18:22:17    File "/sonic/sonic-utilities/scripts/portstat", line 461, in main
18:22:17      portstat.cnstat_diff_print(cnstat_new_dict, cnstat_dict, intf_list, use_json, print_all, errors_only, rates_only)
18:22:17    File "/sonic/sonic-utilities/scripts/portstat", line 297, in cnstat_diff_print
18:22:17      ns_util(cntr.rx_byt, old_cntr.rx_byt, time_gap, port_speed),
18:22:17    File "/sonic/sonic-utilities/utilities_common/netstat.py", line 52, in ns_util
18:22:17      util = rate/(port_rate*1000*1000*1000/8.0)*100
18:22:17  TypeError: unsupported operand type(s) for *: 'NoneType' and 'int'
18:22:17  ____________________ TestPortStat.test_clear_intf_counters _____________________
18:22:17  
18:22:17  self = <tests.portstat_test.TestPortStat object at 0x7f84cca63c50>

Reply:
i do not understand how the error is related to the fix here. previous it is returning either PORT_RATE or int(speed)//1000. after your fix, it is the same.

Reply:
Here "speed" is updated from below line inside for loop:
    speed = self.db.get(self.db.APPL_DB, full_table_id, PORT_SPEED_FIELD)

Previously, the function was returning "int(speed)//1000" (if speed is not None ) or "None"  (if "self.db.get" failed)
With fix, now it returns "PORT_RATE", when "self.db.get" fails.

In latest code, this is fix is already pushed as part of commit "13bd06be696fc14663ddfe5bb6c19abb9b647759". I will rebase and update the PR.

Reply:
thanks for the explanation